src/ptree_readers/read_resource_element/read_resource_element.cxx

```diff
@@ -8,7 +8,7 @@
 // FIXME: Refactor to reduce cyclomatic complexity.
 
 tablator::Resource_Element tablator::ptree_readers::read_resource_element(
-        const boost::property_tree::ptree &resource_tree, bool is_first) {
+        const boost::property_tree::ptree &resource_tree, bool &is_results_resource) {
```

Comment:
Can one not just ask the `Resource_Element` that gets returned by this whether it is a results resource (e.g. by looking for the `type="results"` attribute, or looking at its `Resource_Type`)? Or does that somehow also depend on context not captured by the resource element itself?

Reply:
Yes, but `read_resource_element()` already does the work of deciding whether the Resource_Element it's creating is a `results` one or not.  It basically uses the `without_attrs()` method you worry about below---so right, if a Resource_Element comes in with a Table sub-element but with type other than `results`, someone will be confused.  I'll add a check for that.